boringssl/boringssl_jni/ecrecover.c

```diff
@@ -0,0 +1,159 @@
+#include "ecrecover.h"
+
+#include <openssl/bn.h>
+#include <openssl/ec.h>
+#include <openssl/ecdsa.h>
+#include <openssl/evp.h>
+#include <openssl/mem.h>
+#include <string.h>
+#include <stdio.h>
+
+#define P256_KEY_LEN 65
+#define P256_COORD_LEN 32
+
+int ecrecover_r1(
```

Comment:
This function should be heavily scrutinized. boringssl does not offer public key recovery from signature.  Since we are implementing recovery using boringssl rather than just calling a recovery method, we should scrutinize correctness, edge cases, and performance compared to the openssl implementation.